libs/exec/execv_windows.go

```diff
@@ -0,0 +1,49 @@
+//go:build windows
+
+package exec
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// Note: Windows does not support an execv syscall that replaces the current process.
+// To emulate this, we create a child process, stream the input and output, and return the exit code.
+// ref: https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/execv-wexecv?view=msvc-170
+func execv(opts ExecvOptions) error {
+	path, err := exec.LookPath(opts.Args[0])
+	if err != nil {
+		return fmt.Errorf("looking up %q failed: %w", opts.Args[0], err)
+	}
+
+	cmd := exec.Command(path, opts.Args[1:]...)
+
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	cmd.Dir = opts.Dir
+	cmd.Env = opts.Env
+
+	// Start the child command.
+	err = cmd.Start()
+	if err != nil {
+		return fmt.Errorf(" %s failed: %w", strings.Join(opts.Args, " "), err)
+	}
+
+	err = cmd.Wait()
+	if exitErr, ok := err.(*exec.ExitError); ok {
+		os.Exit(exitErr.ExitCode())
+	}
+	if err != nil {
+		return fmt.Errorf("running %s failed: %w", strings.Join(opts.Args, " "), err)
```

Comment:
it's inconsistent that this function calls os.Exit() on success and on failure but not in this case. Let's change this to print(stderr) + os.Exit as well?

Reply:
That's already the case. If the command fails we exit directly:
```
err = cmd.Wait()
if exitErr, ok := err.(*exec.ExitError); ok {
os.Exit(exitErr.ExitCode())
}
```

Only in cases where cmd.Wait has a failure that is unrelated to the child process do we return control back to the CLI. I think that's the right call since something went wrong trying to run the process rather than with the process itself.